packages/react-instantsearch-dom/src/components/Hits.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { createClassNames } from '../core/utils';
+
+type Hit = {
+  objectID: string | number;
+};
+
+type HitProps = {
+  hit: Hit;
+};
+
+type Props = {
+  hits: Hit[];
+  className?: string;
+  hitComponent?: React.FunctionComponent<HitProps>;
+};
+
+const cx = createClassNames('Hits');
+
+const Hits: React.SFC<Props> = ({
+  hits,
+  className = '',
+  hitComponent: HitComponent = DefaultHitComponent,
+}) => (
+  <div className={classNames(cx(''), className)}>
+    <ul className={cx('list')}>
+      {hits.map(hit => (
+        <li className={cx('item')} key={hit.objectID}>
+          <HitComponent hit={hit} />
+        </li>
+      ))}
+    </ul>
+  </div>
+);
+
+const DefaultHitComponent: React.FC<HitProps> = props => (
+  <div
+    style={{
+      borderBottom: '1px solid #bbb',
+      paddingBottom: '5px',
+      marginBottom: '5px',
+      wordBreak: 'break-all',
+    }}
+  >
+    {JSON.stringify(props).slice(0, 100)}
+    ...
+  </div>
+);
+
+const HitPropTypes = PropTypes.shape({
+  objectID: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
+    .isRequired,
+});
+
+Hits.propTypes = {
```

Comment:
How do we deal with these proptypes redundancy?

Reply:
Right now we type them since we have to use them for non TypeScript users. We can probably generate them automatically with a plugin but for now it's fine to type them twice. TypeScript manage to help you (sometimes) when there is a mismatch between the two.